bot/exts/filtering/filtering.py

```diff
@@ -0,0 +1,1424 @@
+        if result is None:
+            return
+        list_type, filter_list = result
+        await self._add_filter(ctx, noui, list_type, filter_list, content, description_and_settings)
+
+    @filter.command(name="edit", aliases=("e",))
+    async def f_edit(
+        self,
+        ctx: Context,
+        noui: Optional[Literal["noui"]],
+        filter_id: int,
+        *,
+        description_and_settings: Optional[str] = None
+    ) -> None:
+        """
+        Edit a filter specified by its ID.
+
+        Unless `noui` is specified, a UI will be provided to edit the content, description, and settings
+        before confirmation.
+
+        The settings can be provided in the command itself, in the format of `setting_name=value` (no spaces around the
+        equal sign). The value doesn't need to (shouldn't) be surrounded in quotes even if it contains spaces.
+
+        A template filter can be specified in the settings area to copy overrides from. The setting name is "--template"
+        and the value is the filter ID. The template will be used before applying any other override.
+
+        To edit the filter's content, use the UI.
+        """
+        result = self._get_filter_by_id(filter_id)
+        if result is None:
+            await ctx.send(f":x: Could not find a filter with ID `{filter_id}`.")
+            return
+        filter_, filter_list, list_type = result
+        filter_type = type(filter_)
+        settings, filter_settings = filter_serializable_overrides(filter_)
+        description, new_settings, new_filter_settings = description_and_settings_converter(
+            filter_list,
+            list_type, filter_type,
+            self.loaded_settings,
+            self.loaded_filter_settings,
+            description_and_settings
+        )
+
+        content = filter_.content
+        description = description or filter_.description
+        settings.update(new_settings)
+        filter_settings.update(new_filter_settings)
+        patch_func = partial(self._patch_filter, filter_)
+
+        if noui:
+            try:
+                await patch_func(
+                    ctx.message, filter_list, list_type, filter_type, content, description, settings, filter_settings
+                )
+            except ResponseCodeError as e:
+                await ctx.reply(embed=format_response_error(e))
+            return
+
+        embed = Embed(colour=Colour.blue())
+        embed.description = f"`{filter_.content}`"
+        if description:
+            embed.description += f" - {description}"
+        embed.set_author(
+            name=f"Filter {filter_id} - {filter_list[list_type].label}".title())
+        embed.set_footer(text=(
+            "Field names with an asterisk have values which override the defaults of the containing filter list. "
+            f"To view all defaults of the list, "
+            f"run `{constants.Bot.prefix}filterlist describe {list_type.name} {filter_list.name}`."
+        ))
```

Comment:
I feel like it would be nicer to *only* display the overrides in the UI, seeing all the default values feels mostly unnecessary and a bit overwhelming.

An alternative is if it could also be more clever in only showing relevant values, so no empty lists/null values (unless they're overrides). E.g. the `enabled_*` and `disabled_*` lists could be hidden if they're empty, and `infraction_reason`, `infraction_duration`, `infraction_channel` could also be hidden if they're null.

Reply:
I'm a bit conflicted about this. It feels like seeing the full context is important to avoid unexpected behavior. For example having a category disabled by default, and then not figuring out why this isn't working. Or the filter deleting messages by default, and not remembering to switch it off because the value is not in front of you.

The alternative is possible.. I'll have to play with it to see if it makes sense.

Reply:
Hmm, I don't think some settings being missing for the falsey example is too confusing, but lets leave this for now. We can always change things once we have more experience with it in use.